lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart

```diff
@@ -38,9 +45,75 @@ class CreatePostModal extends HookWidget {
   }
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
     final textEditorController = useQuillController();
     final hasContent = useTextEditorHasContent(textEditorController);
+    final hasPoll = useState(false);
+
+    useEffect(
+      () {
+        void checkPoll() {
+          final delta = textEditorController.document.toDelta();
+          var pollExists = false;
+
+          for (final operation in delta.operations) {
+            if (operation.isInsert && operation.data is Map<String, dynamic>) {
+              final data = operation.data! as Map<String, dynamic>;
+              if (data.containsKey('custom')) {
+                final customData = data['custom'];
+                if (customData is Map<String, dynamic> &&
+                    customData.containsKey(textEditorPollKey)) {
+                  pollExists = true;
+                  break;
+                }
+                if (customData is String) {
+                  try {
+                    final parsedData = jsonDecode(customData);
+                    if (parsedData is Map<String, dynamic> &&
+                        parsedData.containsKey(textEditorPollKey)) {
+                      pollExists = true;
+                      break;
+                    }
+                  } catch (e) {
+                    Logger.log('Failed to parse custom data as JSON: $e');
+                  }
+                }
+              }
+            }
+          }
+
+          hasPoll.value = pollExists;
+        }
+
+        textEditorController.addListener(checkPoll);
+
+        checkPoll();
+
+        return () {
+          textEditorController.removeListener(checkPoll);
+        };
+      },
+      [textEditorController],
+    );
+
+    bool isPollValid() {
+      final pollTitle = ref.watch(pollTitleNotifierProvider);
```

Comment:
```
  final pollTitle = ref.watch(pollTitleNotifierProvider);
  final pollAnswers = ref.watch(pollAnswersNotifierProvider);
```

change to `ref.read`

Reply:
It's used inside callback, so it should be called with `ref.read` or you can get values with `ref.watch`, but then you need to move these lines outside of the `isPollValid` method